src/drivers/ins/microstrain/MicroStrain.cpp

```diff
@@ -0,0 +1,623 @@
+{
+	MicroStrain *ref = static_cast<MicroStrain *>(user);
+
+	if (mip_packet_descriptor_set(packet) != MIP_SENSOR_DATA_DESC_SET) {
+		return;
+	}
+
+	mip_sensor_scaled_accel_data accel;
+	mip_sensor_scaled_gyro_data gyro;
+	mip_sensor_scaled_mag_data mag;
+	mip_sensor_scaled_pressure_data baro;
+
+	bool accel_valid = false;
+	bool gyro_valid = false;
+	bool mag_valid = false;
+	bool baro_valid = false;
+
+	auto t = hrt_absolute_time();
+
+	for (mip_field field = mip_field_first_from_packet(packet); mip_field_is_valid(&field); mip_field_next(&field)) {
+		switch (mip_field_field_descriptor(&field)) {
+		case MIP_DATA_DESC_SENSOR_ACCEL_SCALED: extract_mip_sensor_scaled_accel_data_from_field(&field, &accel);
+			accel_valid = true; break;
+
+		case MIP_DATA_DESC_SENSOR_GYRO_SCALED: extract_mip_sensor_scaled_gyro_data_from_field(&field, &gyro); gyro_valid = true;
+			break;
+
+		case MIP_DATA_DESC_SENSOR_MAG_SCALED: extract_mip_sensor_scaled_mag_data_from_field(&field, &mag); mag_valid = true;
+			break;
+
+		case MIP_DATA_DESC_SENSOR_PRESSURE_SCALED: extract_mip_sensor_scaled_pressure_data_from_field(&field, &baro);
+			baro_valid = true; break;
+
+		default: break;
+
+		}
+	}
+
+
+	if (accel_valid) {
+		ref->_px4_accel.update(t, accel.scaled_accel[0]*CONSTANTS_ONE_G,
+				       accel.scaled_accel[1]*CONSTANTS_ONE_G,
+				       accel.scaled_accel[2]*CONSTANTS_ONE_G);
+
+	}
+
+	if (gyro_valid) {
+		ref->_px4_gyro.update(t, gyro.scaled_gyro[0],
+				      gyro.scaled_gyro[1],
+				      gyro.scaled_gyro[2]);
+	}
+
+	if (mag_valid) {
+		ref->_px4_mag.update(t, mag.scaled_mag[0],
+				     mag.scaled_mag[1],
+				     mag.scaled_mag[2]);
+	}
+
+	if (baro_valid) {
+		ref->_sensor_baro.timestamp = timestamp;
+		ref->_sensor_baro.timestamp_sample = t;
+		ref->_sensor_baro.pressure = baro.scaled_pressure * 100.f; // convert [Pa] to [mBar]
+		ref->_sensor_baro_pub.publish(ref->_sensor_baro);
+
+	}
+
+}
+
+bool MicroStrain::init()
+{
+	// Run on fixed interval
+	ScheduleOnInterval(ms_schedule_rate_us);
+
+	return true;
+}
+
+
+void MicroStrain::Run()
+{
+	if (should_exit()) {
+		ScheduleClear();
+		exit_and_cleanup();
+		return;
+	}
+
+	perf_begin(_loop_perf);
+	perf_count(_loop_interval_perf);
+
+	initialize_ins();
+
+	// Initialization failed, stop the module
+	if (_is_init_failed) {
+		request_stop();
+		perf_end(_loop_perf);
+		return;
+	}
+
+	// Check if parameters have changed
+	if (_parameter_update_sub.updated()) {
+		// clear update
+		parameter_update_s param_update;
+		_parameter_update_sub.copy(&param_update);
+		updateParams(); // update module parameters (in DEFINE_PARAMETERS)
+
+	}
+
+	mip_interface_update(&device, false);
+
+	perf_end(_loop_perf);
+}
+
+int MicroStrain::task_spawn(int argc, char *argv[])
+{
+	int ch;
+	int myoptind = 1;
+	const char *myoptarg = nullptr;
+
+	const char *dev = "/dev/ttyS4";
+	int32_t rot = ROTATION_NONE;
+
+	while ((ch = px4_getopt(argc, argv, "d:r:", &myoptind, &myoptarg)) != EOF) {
```

Comment:
I see that vectornav does the same thing here. I would prefer to read all information in from parameters instead of from the command line